vowpalwabbit/cb_explore_adf_greedy.cc

```diff
@@ -67,6 +65,19 @@ void cb_explore_adf_greedy::predict_or_learn_impl(LEARNER::multi_learner& base,
     preds[0].score += 1.f - _epsilon;
 }
 
+
+template <bool is_learn>
+void cb_explore_adf_greedy::predict_or_learn_impl(LEARNER::multi_learner& base, multi_ex& examples)
+{
+  // Explore uniform random an epsilon fraction of the time.
+  if(is_learn)
```

Comment:
Tried this out.  However doing the following: 

if VW_STD17_CONSTEXPR (is_learn)

Seems to make code unnecassarily cryptic.  Release mode with optimization properly compiles this out in msvc and gcc.  If adding VW_STD17_CONSTEXPR makes an expression more efficient, it would make sense to add it then.

Reply:
No problem, sounds good.